### PR TITLE
[NEP-15392] Internationalise - Notes sidebar section

### PR DESCRIPTION
**Pull request by lincsanders**
Ticket: https://jobready.atlassian.net/browse/NEP-15392

Linked to: https://github.com/rdytech/neptune/pull/8799

*What does this PR do?*

- ~Added opt-in translation configuration for labels / buttons.~
- ~Bumps version from 2.3.x to 2.4.0~

- Translations assume a high chance of pre-existing RoR environment with global I18n JS object and try to pull from that
- New translation default english values for buttons
- t() tags implemented
- All free hanging text goes through `I18n.t` now.
- Namespaces filterbar translations for cleaner locale files everywhere.
- Logic change for cases where placeholders (now localized) were being used as dynamic keys on eventual objects

image

### CODE REVIEW

**Comment by JordanMooreNZ** [comment]
I notice you have seen the other [PR](https://github.com/rdytech/react-filterbar/pull/129) doing similar stuff, any particular reason you want to use another approach rather than using i18n-js like the other PR? Feels like it would be easier to be consistent in our usage i.e using something like
```
t('filterbar.apply')
```
over
```
{this.context.filterBarStore?.localizations?.apply || 'Apply'}
```

**Comment by lincsanders** [comment]
> I notice you have seen the other [PR](https://github.com/rdytech/react-filterbar/pull/129) doing similar stuff, any particular reason you want to use another approach rather than using i18n-js like the other PR? Feels like it would be easier to be consistent in our usage i.e using something like
> 
> ```
> t('filterbar.apply')
> ```
> 
> over
> 
> ```
> {this.context.filterBarStore?.localizations?.apply || 'Apply'}
> ```

Nah, this is definitely a much nicer way... I'll adapt the other PR's methodology to this one instead.

**Comment by lincsanders** [comment]
@JordanMooreNZ hella better implementation now.

File: src/components/FilterBar/FilterList/FilterList.react.js
Comment (JordanMooreNZ) [inline src/components/FilterBar/FilterList/FilterList.react.js]: Is this still required?

**Review by JordanMooreNZ — COMMENTED** [review]
Small comment otherwise looks pretty good 🙌 we'll need a version update though to `2.3.2.rc1`, then we can test it on the Neptune side before updating to the official `2.3.2` for both Neptune and here

File: src/components/FilterBar/FilterList/FilterList.react.js
Comment (lincsanders) [reply src/components/FilterBar/FilterList/FilterList.react.js]: Absolutely not, just my terrible refactoring skillz, removed.

**Review by JordanMooreNZ — COMMENTED** [review]
LGTM ✅ will hold off on approving until its the official version after the Neptune side of this goes through initial QA with the rc1 version

**Comment by lincsanders** [comment]
> LGTM ✅ will hold off on approving until its the official version after the Neptune side of this goes through initial QA with the rc1 version

Bumped to rc2, all free hanging strings should now be translated so this might help for other tickets 🤔

File: src/components/Table/Pagination.react.js
Comment (caothanh510) [inline src/components/Table/Pagination.react.js:60]: can you please recheck the format with this PR https://github.com/rdytech/neptune/pull/8727/files#diff-a5a86160b667645873b275b84f4524f4ade7347bbbd25813378cb148d0826be3R1350-R1352

Comment (caothanh510) [reply src/components/Table/Pagination.react.js:60]: ah all good I saw it updated in your PR https://github.com/rdytech/neptune/pull/8799

**Comment by lincsanders** [comment]
@JordanMooreNZ @caothanh510 I think we should finalise this (remove RC) with a proper version now, we've had RC running in prod, let's do a bump to 2.3.2 and put a bow on this PR.

**Comment by caothanh510** [comment]
> @JordanMooreNZ @caothanh510 I think we should finalise this (remove RC) with a proper version now, we've had RC running in prod, let's do a bump to 2.3.2 and put a bow on this PR.

yeah i think we should remove RC after this PR merge https://github.com/rdytech/neptune/pull/8878

**Review by caothanh510 — APPROVED** [review]
LGTM, nice work 💯

**Pull request merged**
